samples/Samples.Jwt/JwtWebSocketAuthenticationService.cs

```diff
@@ -29,25 +44,21 @@ public Task AuthenticateAsync(IWebSocketConnection connection, string subProtoco
             if (connection.HttpContext.User.Identity?.IsAuthenticated ?? false)
                 return Task.CompletedTask;
 
-            // attempt to read the 'Authorization' key from the payload object and verify it contains "Bearer: XXXXXXXX"
+            // attempt to read the 'Authorization' key from the payload object and verify it contains "Bearer XXXXXXXX"
             var authPayload = _graphQLSerializer.ReadNode<AuthPayload>(operationMessage.Payload);
             if (authPayload != null && authPayload.Authorization != null && authPayload.Authorization.StartsWith("Bearer ", StringComparison.Ordinal))
             {
                 // pull the token from the value
                 var token = authPayload.Authorization.Substring(7);
-                // parse the token in the same manner that the .NET AddJwtBearer() method does
-                var handler = new Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler();
-                var result = handler.ValidateToken(token, JwtHelper.Instance.TokenValidationParameters);
-                if (result.IsValid)
-                {
-                    // convert JWT tokens with "role" as the claim type (a JWT defined claim type) to the proper .NET claim type constant http://schemas.microsoft.com/ws/2008/06/identity/claims/role
-                    // note this conversion automatically happens within the .NET AddJwtBearer() method when authenticating GET/POST requests
-                    var claims = result.ClaimsIdentity.Claims.Select(claim => claim.Type == "role" ? new Claim(result.ClaimsIdentity.RoleClaimType, claim.Value) : claim);
-                    var principal = new ClaimsPrincipal(new ClaimsIdentity(claims, JwtBearerDefaults.AuthenticationScheme));
-
-                    // set the ClaimsPrincipal for the HttpContext; authentication will take place against this object
-                    connection.HttpContext.User = principal;
-                }
+                // parse the token in the same manner that the .NET AddJwtBearer() method does:
+                // JwtSecurityTokenHandler maps the 'name' and 'role' claims to the 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name'
+                // and 'http://schemas.microsoft.com/ws/2008/06/identity/claims/role' claims;
+                // this mapping is not performed by Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler
+                var handler = new System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler();
+                var tokenValidationParameters = _jwtBearerOptionsMonitor.Get(JwtBearerDefaults.AuthenticationScheme).TokenValidationParameters;
+                var principal = handler.ValidateToken(token, tokenValidationParameters, out var securityToken);
```

Comment:
No `TokenValidationResult` anymore? Does `ValidateToken` throw in case of any error?

Reply:
Correct; it throws in the case that the token is invalid for any reason (according to the parameters defined in `tokenValidationParameters`). So for example, if `TokenValidationParameters.ValidateIssuer` is `true`, then the issuer must match or an exception is thrown.